src/accountsdb/buffer_pool.zig

```diff
@@ -735,190 +757,356 @@ pub const HierarchicalFIFO = struct {
 
 /// slice-like datatype
 /// view over one or more buffers owned by the BufferPool
-pub const CachedRead = struct {
-    buffer_pool: *BufferPool,
-    frame_indices: []const FrameIndex,
-    /// inclusive, the offset into the first frame
-    first_frame_start_offset: FrameOffset,
-    /// exclusive, the offset into the last frame
-    last_frame_end_offset: FrameOffset,
+pub const ReadHandle = union(enum) {
+    /// Data owned by BufferPool, returned by .read() - do not construct this yourself (!)
+    cached: CachedRead,
+
+    /// Data allocated elsewhere, not owned or created by BufferPool. BufferPool will deallocate.
+    owned_allocation: []const u8,
+    /// Data allocated elsewhere, not owned or created by BufferPool.
+    unowned_allocation: []const u8,
+    /// Data owned by parent ReadHandle
+    sub_read: SubRead,
+    /// Used in place of a read, in callsites where it is not actually needed. Provides .len().
+    unread: Unread,
```

Comment:
empty?

Reply:
renamed to empty f40689214b7f8a0945d24c5a2847c29af6db0203